docs/tutorial/reconstruction_system/make_fragments.rst

```diff
@@ -74,24 +74,13 @@ Make a fragment
 
 .. literalinclude:: ../../../examples/python/reconstruction_system/make_fragments.py
    :language: python
-   :lineno-start: 124
-   :lines: 27,125-146
+   :pyobject: integrate_rgb_frames_for_fragment
    :linenos:
+   :lineno-match:
 
-Once the poses are estimates, :ref:`/tutorial/pipelines/rgbd_integration.ipynb`
+Once the poses are estimated, :ref:`/tutorial/pipelines/rgbd_integration.ipynb`
 is used to reconstruct a colored fragment from each RGBD sequence.
 
-Batch processing
-``````````````````````````````````````
-
-.. literalinclude:: ../../../examples/python/reconstruction_system/make_fragments.py
-   :language: python
-   :lineno-start: 181
-   :lines: 27,182-205
-   :linenos:
-
-The main function calls each individual function explained above.
-
```

Comment:
@ssheorey The reason I removed this is because the `run` method calls only `process_single_fragment`, however `process_single_fragment` is never discussed in documentation before. Also, `run` method focuses on use of multiprocessing to speedup the code, but thats not really where open3d documentation focus is.

I dived deeper into git history and originally `process_single_fragment` used to be `process_fragments` which did all prep before calling previous functions for each fragment. It was documentated in Batch Processing section - https://github.com/isl-org/Open3D/pull/521/files.

See http://www.open3d.org/docs/0.6.0/tutorial/ReconstructionSystem/make_fragments.html#batch-processing for easy check (although code lines went a little out of sync with docs here)

See http://www.open3d.org/docs/0.7.0/tutorial/ReconstructionSystem/make_fragments.html#batch-processing where this got changed to `run` method.

Will it better to revert my change but instead we show either  

- `process_single_fragment`. _OR_
- both `process_single_fragment` and `run` method. This would require adding a few statements explaining use of multiprocessing also.

Reply:
Option 2 sounds good. Let's show both `process_single_fragment` and `run` and explain use of multiprocessing to speed up processing. Optionally the parallel execution in `run()` can be cleaned up in the example. `joblib` can be replaced with `concurrent.futures.ProcessPoolExecutor` or `multiprocessing.pool`, which is part of the Python standard library.

Reply:
@ssheorey Need your input. I tried replacing joblib, but in an earlier problem at work, I had faced similar issue where open3d hangs with multiprocessing. As per your comment here, https://github.com/isl-org/Open3D/issues/1552#issuecomment-1009478420, I moved to forkserver and that is working fine. Shall I go with this?

```
max_workers = min(1, multiprocessing.cpu_count()-1, n_fragments)
mp_context = multiprocessing.get_context('forkserver')
with mp_context.Pool(processes=max_workers) as pool:
    args = [(fragment_id, color_files, depth_files, n_files, n_fragments, config) for fragment_id in range(n_fragments)]
    pool.starmap(process_single_fragment, args)
```

P.S. Was curious, and found joblib went with forkserver for same reason but later reverted back to `fork` -

Reply:
Yes, that's a good idea. This will also be a reference to users about how to use Open3D with multiprocessing.
Just add a comment saying "Linux default method 'fork' cannot be used with multithreaded Open3D."

Reply:
python_mp_spawn is using spawn as starting method.
python_mp_forkserver is using forkserver as starting method.
python_joblib_mp_backend is using joblib with multiprocessing backend.
python_joblib_mp_backend is using joblib with loky backend.
python_single_process is just main process.

Only for 4 fragments with each fragment made of 10 images, I get following measurements:-

```
Code block 'python_mp_spawn' took:            79258.28545 ms
Code block 'python_mp_forkserver' took:       81994.75125 ms
Code block 'python_joblib_mp_backend' took:   92798.19047 ms
Code block 'python_joblib_loky_backend' took:  8461.00342 ms
Code block 'python_single_process' took:      28939.51906 ms
```

```
# Without over allocation of openmp threads, See https://github.com/joblib/joblib/pull/940
os.environ['OMP_NUM_THREADS'] = str((multiprocessing.cpu_count() - 1) // max_workers)
Code block 'python_mp_spawn' took:            7003.84680 ms
Code block 'python_mp_forkserver' took:       7078.38375 ms
Code block 'python_joblib_mp_backend' took:   8058.99887 ms
Code block 'python_joblib_loky_backend' took: 7350.69630 ms
Code block 'python_single_process' took:     30476.80167 ms
```

```
# I also tested with OMP_NUM_THREADS=1. Very slightly slower.
Code block 'python_mp_spawn' took:            7040.57154 ms
Code block 'python_mp_forkserver' took:       7271.84740 ms
Code block 'python_joblib_mp_backend' took:   8278.31439 ms
Code block 'python_joblib_loky_backend' took: 7983.64183 ms
Code block 'python_single_process' took:     31091.80214 ms
```

Didnt make much of a difference, despite with 16 cores and 4 child processes. As this is an introductory example, just setting 1 would be simpler and easier for new users IMO. 


Final committed code test

Multiprocess takes    `Making fragments    0:05:11.274633` (15 cores)
Singleprocess takes  `Making fragments    0:32:18.475112` (OpenMP number of threads unchanged)